File: Android/AndroidStudioProjects/TesJava/app/src/main/java/com/samzuhalsetiawan/tesjava/MainActivity.java
package com.samzuhalsetiawan.tesjava;

import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private final ActivityResultLauncher<Intent> launcher =
            registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
                Intent intent = result.getData();
                //TODO: Do Something with the result

            });

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button mButton = findViewById(R.id.btnStartActivity);
        mButton.setOnClickListener(v -> {
            launcher.launch(new Intent(MainActivity.this, SecondActivity.class));
        });
    }
}

File: Android/AndroidStudioProjects/TesKotlin/app/src/main/java/com/samzuhalsetiawan/teskotlin/MainActivity.kt
package com.samzuhalsetiawan.teskotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import androidx.activity.result.contract.ActivityResultContracts

class MainActivity : AppCompatActivity() {

    private val launcher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
        it.data?.getStringExtra("EXTRA_DATA").also { println(it) }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.btnStartActivity).setOnClickListener {
            launcher.launch(Intent(this, SecondActivity::class.java))
        }

    }
}